### PR TITLE
WIP Discovery Endpoints

### PR DESCRIPTION
**Pull request by aphelionz**
Putting this in for feedback knowing one of my tests will fail... would love some insight as to why - the Subject filtering should work fine, as should the split function.

### CODE REVIEW

File: hs_core/tests/api/rest/test_resource_list.py [hunk 1]
```diff
@@ -184,3 +184,28 @@ def test_resource_list_by_type(self):
...
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        content = json.loads(response.content)
+        self.assertEqual(content['count'], 1)
+
+        response = self.client.get('/hsapi/resource/', {'subject': 'other'}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        content = json.loads(response.content)
+        self.assertEqual(content['count'], 1)
+
+        response = self.client.get('/hsapi/resource/', {'subject': 'both,other'}, format='json')
```
Comment (pkdash) [inline hs_core/tests/api/rest/test_resource_list.py]: Should you be passing here ```{'subject': 'one,other'}```? to have to count to be 2

Comment (aphelionz) [inline hs_core/tests/api/rest/test_resource_list.py]: ahhh, yes, thank you. I knew it was something dumb

File: hs_core/hydroshare/users.py [hunk 2]
```diff
@@ -238,19 +238,10 @@ def get_public_groups():
...
-        full_text_search=None,
-        published=False,
-        edit_permission=False,
-        public=False,
-        type=None,
-        author=None,
-        contributor=None,
-        subject=None,
-):
+def get_resource_list(creator=None, group=None, user=None, owner=None, from_date=None,
```
Comment (pkdash) [inline hs_core/hydroshare/users.py]: @aphelionz  I am trying to find out how this list gets generated based on bounding box information.

**Comment by aphelionz** [comment]
@pkdash Please review again :)

File: hs_core/hydroshare/users.py [hunk 3]
```diff
@@ -277,6 +279,11 @@ def get_resource_list(creator=None, group=None, user=None, owner=None, from_date
         count = int
         subject = list of subject
         type = list of resource type names, used for filtering
+        coverage_type = geo parameter, one of box or point
```
Comment (pkdash) [inline hs_core/hydroshare/users.py]: It seems ```coverage_type``` can be anything other than ```None```

File: hs_core/hydroshare/users.py (hunk 2 above)
Comment (aphelionz) [inline hs_core/hydroshare/users.py]: @pkdash should be evident now

File: hs_core/tests/api/rest/test_resource_list.py (hunk 1 above)
Comment (aphelionz) [inline hs_core/tests/api/rest/test_resource_list.py]: @pkdash should be fixed

File: hs_core/hydroshare/users.py (hunk 3 above)
Comment (aphelionz) [inline hs_core/hydroshare/users.py]: Fixed by changing it to `if coverage_type in ('box', 'point'):`

File: hs_core/tests/api/rest/test_resource_list.py [hunk 4]
```diff
@@ -211,46 +211,55 @@ def test_resource_list_by_keyword(self):
         self.assertEqual(content['count'], 2)
 
     def test_resource_list_by_bounding_box(self):
```
Comment (pkdash) [inline hs_core/tests/api/rest/test_resource_list.py]: Would be good to include a test for bad request using parameter ```coverage_type```.

File: hs_core/tests/api/rest/test_resource_list.py [hunk 5]
```diff
@@ -263,3 +263,11 @@ def test_resource_list_by_bounding_box(self):
         self.assertEqual(response.status_code, status.HTTP_200_OK)
         content = json.loads(response.content)
         self.assertEqual(content['count'], 3)
+
+        # Bad coverage has no effect
+        response = self.client.get('/hsapi/resource/', {'coverage_type': 'bad',
+                                                        'nonsensical': '90',
+                                                        'params': '140'}, format='json')
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```
Comment (pkdash) [inline hs_core/tests/api/rest/test_resource_list.py]: Should the api return here status code 400?

Comment (aphelionz) [inline hs_core/tests/api/rest/test_resource_list.py]: It doesn't return a 400 on any other nonsensical params. Open a separate issue?

Comment (pkdash) [inline hs_core/tests/api/rest/test_resource_list.py]: I am wondering if we change the serializer like this we may not have to change anything else. The api then should return 400 for the bad test case.
```coverage_type = serializers.CharField(required=False) ``` -> ```coverage_type = serializers.ChoiceField(choices=['box', 'point'], required=False)```

Comment (aphelionz) [inline hs_core/tests/api/rest/test_resource_list.py]: That seemed to work. Pushed a new commit @pkdash

**Comment by pkdash** [comment]
+1. Can merge after Jenkins checks are successful.

**Comment by pkdash** [comment]
@aphelionz Would be good to get this merged soon.

**Pull request merged**